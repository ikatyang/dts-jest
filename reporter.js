module.exports = require('./lib/index').Reporter;
