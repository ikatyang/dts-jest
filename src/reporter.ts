import * as path from 'path';
import { config_namespace, JestConfig } from './definitions';
import { get_config } from './utils/get-config';

export class Reporter implements jest.Reporter {
  // tslint:disable-next-line:naming-convention prefer-function-over-method
  public onRunComplete(contexts: Set<jest.Context>) {
    process.stdout.write('\n');

    contexts.forEach(context => {
      const {
        typescript: { version: ts_version },
        typescript_path: ts_path,
      } = get_config(
        (context.config as JestConfig).globals[config_namespace],
        context.config.rootDir,
      );

      // istanbul ignore next
      const context_name =
        contexts.size > 1 ? ` ${path.basename(context.config.rootDir)}` : '';

      process.stdout.write(
        `[dts-jest]${context_name} using TypeScript v${ts_version} from ${ts_path}\n`,
      );
    });

    process.stdout.write('\n');
  }
}
