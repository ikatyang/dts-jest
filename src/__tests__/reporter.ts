import { Reporter } from '../reporter';

// tslint:disable-next-line:no-unbound-method
const original_stdout_write = process.stdout.write;
let mocked_stdout_write: jest.MockInstance<any>;

beforeEach(() => {
  mocked_stdout_write = process.stdout.write = jest.fn();
});

afterAll(() => {
  process.stdout.write = original_stdout_write;
});

it('should report correctly', () => {
  const reporter = new Reporter(create_config());
  reporter.onRunComplete(new Set([create_context()]));
  expect(get_write_content()).toMatchSnapshot();
});

function create_context(): jest.Context {
  return {
    config: {
      rootDir: process.cwd(),
      globals: {},
    },
  } as any;
}

function create_config(): jest.GlobalConfig {
  return {
    useStderr: false,
  } as any;
}

function get_write_content() {
  return mocked_stdout_write.mock.calls
    .map(x => x.join(''))
    .join('')
    .replace(process.cwd(), '<cwd>');
}
