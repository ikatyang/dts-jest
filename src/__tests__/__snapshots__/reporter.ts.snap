// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should report correctly 1`] = `
"
,[dts-jest] using TypeScript v2.4.2 from <cwd>/node_modules/typescript/lib/typescript.js
,
"
`;
